metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+            default=None,
+            help="Specify run pathspec for testing @trigger_on_finish locally.",
+        ),
+    }
+
+    def flow_init(
+        self,
+        flow_name,
+        graph,
+        environment,
+        flow_datastore,
+        metadata,
+        logger,
+        echo,
+        options,
+    ):
+        self.triggers = []
+        if sum(map(bool, (self.attributes["flow"], self.attributes["flows"]))) > 1:
+            raise MetaflowException(
+                "Specify only one of *flow* or *flows* "
+                "attributes in *@trigger_on_finish* decorator."
+            )
+        elif self.attributes["flow"]:
+            # flow supports the format @trigger_on_finish(flow='FooFlow')
+            if is_stringish(self.attributes["flow"]):
+                self.triggers.append(
+                    {
+                        "fq_name": self.attributes["flow"],
+                    }
+                )
+            else:
+                raise MetaflowException(
+                    "Incorrect type for *flow* attribute in *@trigger_on_finish* "
+                    " decorator. Supported type is string - \n"
+                    "@trigger_on_finish(flow='FooFlow')"
+                )
+        elif self.attributes["flows"]:
+            # flows attribute supports the following formats -
+            #     1. flows=['FooFlow', 'BarFlow']
+            if isinstance(self.attributes["flows"], list):
+                for flow in self.attributes["flows"]:
+                    if is_stringish(flow):
+                        self.triggers.append(
+                            {
+                                "fq_name": flow,
+                            }
+                        )
+                    else:
+                        raise MetaflowException(
+                            "One or more flows in *flows* attribute in "
+                            "*@trigger_on_finish* decorator have an incorrect type. "
+                            "Supported type is string - \n"
+                            "@trigger_on_finish(flows=['FooFlow', 'BarFlow']"
+                        )
+            else:
+                raise MetaflowException(
+                    "Incorrect type for *flows* attribute in *@trigger_on_finish* "
+                    "decorator. Supported type is list - \n"
+                    "@trigger_on_finish(flows=['FooFlow', 'BarFlow']"
+                )
+
+        if not self.triggers:
+            raise MetaflowException(
+                "No flow(s) specified in *@trigger_on_finish* decorator."
+            )
+
+        # Make triggers @project aware
+        for trigger in self.triggers:
+            if trigger["fq_name"].count(".") == 0:
+                # fully qualified name is just the flow name
+                trigger["flow"] = trigger["fq_name"]
+            elif trigger["fq_name"].count(".") in (2, 3):
+                # fully qualified name is of the format - project.branch.flow_name
```

Comment:
clarify the two formats for clarity:
`project.prod.flow_name` or `project.[user|prod|test].branch_name.flow_name`.